duckduckgo/iOS
# NetP Invite code screen PR 2

Task/Issue URL: https://app.asana.com/0/0/1205084446087083/f
BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/441
CC:

**Description:**

Second PR for the NetP Invite Code flow. You can find designs in the linked task above. This builds upon #1881 and adds to the existing Invite Code presentation logic in the `NetworkProtectionInviteViewModel`. It also fleshes out and designs the `NetworkProtectionInviteView` which has two states:

- codeEntry
- success

This also adds some tests which required some mocks / test helpers for types in the NetworkProtection library in BSK. As we will likely need these on macOS when we come to add more tests there, I added a NetworkProtectionTestUtils library to BSK. This PR points to that BSK branch so I will make sure to point to the BSK release once that’s approved. BSK PR linked above.

**Steps to test this PR:**

1. Check out this branch
2. Run the app
3. Ensure you’re authenticated as internal
4. Navigate to the Settings
5. Select Network Protection
6. If you’ve already enabled it, tap the button to clear the invite code then navigate back and select the NetP item again
7. Enter a random string and submit it to trigger the error label.
8. Now enter an invite code from [Network Protection Invite Codes](https://app.asana.com/0/1203135672790568/1204303483337129/f) and submit it
9. The success screen will be shown. Click Get Started to launch the status view

**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
